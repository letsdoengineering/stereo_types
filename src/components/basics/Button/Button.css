.basic_button {
    font-weight: 700;
    font-size: 1rem;
    background-color: var(--default-blue);
    color: white;
    border: 3px solid var(--default-blue);
    padding: 8px;
}

.basic_button_fill-space {
    height: 100%;
    width: 100%;
    font-size: 1.2rem;
}

.basic_button_secondary {
    background: transparent;
    color: var(--default-blue);
}

.basic_button:hover {
    cursor: pointer;
    outline:  white solid 5px;
}

.basic_button:focus {
    outline:  white solid 5px;
}

.basic_button_disabled {
    background: var(--slate-gray);
    color: var(--dark-gray);
    border: 3px solid transparent;
}

.basic_button_disabled:hover {
    cursor: not-allowed;
    color: black;
    background: gray;
    border: 3px solid transparent;
}


.basic_button_size-S {
    padding: 8px 16px;
}

.basic_button_size-M {
    padding: 24px 32px
}
