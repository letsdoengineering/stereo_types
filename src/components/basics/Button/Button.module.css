.button {
    font-weight: 700;
    font-size: 1rem;
    background-color: var(--default-blue);
    color: white;
    border: 3px solid var(--default-blue);
}

.button-fill-space {
    height: 100%;
    width: 100%;
    font-size: 1.2rem;
}

.button-secondary {
    background: transparent;
    color: var(--default-blue);
}

.button:hover {
    cursor: pointer;
    color: white;
    background:  var(--light-blue);
    border: 3px solid var(--light-blue);
}

.button:focus {
    color: white;
    background:  var(--light-blue);
    border: 3px solid var(--light-blue);
}

.button-disabled {
    background: gray;
    color: black;
    border: 3px solid transparent;
}

.button-disabled:hover {
    cursor: not-allowed;
    color: black;
    background: gray;
    border: 3px solid transparent;
}


.size-S {
    padding: 0.5rem 1rem;
}

.size-M {
    padding: 1.5rem 2rem
}
