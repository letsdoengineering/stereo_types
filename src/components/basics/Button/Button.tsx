import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import buttonStyles from './Button.module.css'

type ButtonProps = {
  /** disabled */
  disabled?: boolean
  /** type of button can only be button or submit - reset type not allowed */
  type?: 'button' | 'submit'
  /** text to display on the button */
  buttonText: string
  /** a standardised size of the button */
  size?: 'S' | 'M'
  /** style type */
  secondary?: boolean
  /** grow to fill container */
  fillSpace?: boolean
  /** onClick callback */
  onClick?(): void
  /** class names represented as a string */
  className?: string
}

const Button: React.FC<ButtonProps> = ({
  disabled = false,
  className,
  buttonText,
  fillSpace = false,
  type = 'submit',
  size = 'S',
  onClick,
  secondary = false,
  ...rest
}: ButtonProps) => {
  const classNames = getClassNames({
    defaultClasses: [`${buttonStyles.button}`, buttonStyles[`size${size}`]],
    className,
    conditionalClasses: {
      [`${buttonStyles.buttonSecondary}`]: secondary,
      [`${buttonStyles.buttonFillSpace}`]: fillSpace,
      [`${buttonStyles.buttonDisabled}`]: disabled,
    },
  })

  return (
    <button disabled={disabled} type={type} className={classNames} onClick={onClick} {...rest}>
      {buttonText}
    </button>
  )
}

export default Button
