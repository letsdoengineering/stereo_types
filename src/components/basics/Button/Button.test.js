import * as React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'
import Button from './Button'

describe('Button', () => {
    it('has not changed the JSX layout of the page from previous snapshot', () => {
        const tree = renderer.create(<Button buttonText='some button' />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders the page DOM that is accessible', async () => {
        const { container } = render(<Button buttonText='some button' />)
        const results = await axe(container)

        expect(results).toHaveNoViolations()
    })
})
