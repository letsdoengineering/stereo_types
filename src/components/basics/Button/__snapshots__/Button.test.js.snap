// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has not changed the JSX layout of the page from previous snapshot 1`] = `
<button
  className="button sizeS buttonFillSpace"
  type="submit"
>
  some button
</button>
`;
